### ISSUE TITLE
Color contrast to low for tables with a background color set

### ISSUE THREAD

**Issue by 2ndkauboy** [WordPress/twentytwentyone#226]
**Describe the bug**

This is a similar issue to #185

All background color that can be chosen from the table block are light. Currently with a dark background the theme is using a light color for the text. This results in unreadable text with almost any available background color.

There is a second issue with tables using the stripes style and having a background color set, showing an additional border with the color of the background on the bottom of the last row and on the right side of the table. This also happens on light page backgrounds.

**To Reproduce**

1. Go to the customizer and change the body background color to black
2. Create or open a post or page.
3. Add a table block with some content.
4. Select a background color for the table.
5. See that the text color is white on a light background and difficult to read.

**Screenshots**

![table-dark-background-before](https://user-images.githubusercontent.com/1793177/94857195-6a675e80-0431-11eb-8bb9-a8e0570253de.PNG)
![table-light-background-before](https://user-images.githubusercontent.com/1793177/94857209-6dfae580-0431-11eb-95fc-25cced3f36f5.PNG)
